### ISSUE TITLE
`scipy-stubs`

### ISSUE THREAD

**Issue by jorenham** [BarakKatzir/types-scipy-sparse#10]
Just wanted to make you aware of [`scipy-stubs`](https://github.com/jorenham/scipy-stubs), as it overlaps with `types-scipy-sparse`. 

In https://github.com/scipy/scipy/issues/21614 I explained why I chose to go about it this way.

At the moment, the `scipy.sparse` part of `scipy-stubs` is very bare-bones, and from the looks of it, some of the work done in `types-scipy-sparse` could help out with that.

So are you perhaps willing to join forces on this one? I'm not all to familiar myself with `scipy.sparse`, so your expertise would be very welcome!

**Comment by BarakKatzir** [comment]
I was not aware of [`scipy-stubs`](https://github.com/jorenham/scipy-stubs) and am pretty sure it and the issue that you linked to didn't exist when I opened my repo. If it did I would have contacted you to join forces! 

The main reason I opened `types-scipy-sparse` is to remove a bunch of type annotation files that I wrote that I found myself carrying from project to project.

The only thing I'm worried about is whether distributing the experimental stubs in a single package is preferable over distributing them in several packages - that way a user can choose which experimental stubs they want for which scipy module.
This splitting to many packages (or a namespace package if that's possible for a stubs-only package) can affect a user in three ways I can think of:

1. A user can pick which stub package to use. There aren't a lot of scipy stubs available publicly (I only know of is [`python-type-stubs`](https://github.com/microsoft/python-type-stubs/tree/main/stubs/scipy-stubs) by microsoft which is also bare-bones) to choose from, but some people probably have their own private partial stubs files or packages. As far as I know mypy doesn't allow a user to pick which stubs to user from which stub package / file for which module (the order of resolution described in PEP-561 doesn't leave such user freedom to my understanding).

2. Another scenario I'm worried about is how much work needs to go into writing and maintaining good type annotations for all of scipy. If the type annotations are good for one module but subpar for another it might deter a user from using the whole package. I'm not sure about other modules, but writing good **generic** stubs for the `sparse` module is very intricate and I probably missed a few edge cases.

3. Finally, if and when scipy maintainers decide to have official annotations they will probably do that in steps covering different parts of scipy at different stages. At that point a user might want to pick the official stubs over the experimental stubs, but still use the experimental stubs where there are no official stubs. But, this point is probably me thinking a bit too far ahead.

All these being said, I think joining forces is the fastest way to get more static typing tools applicable to scipy. The more people who care working together on this the better :)

On a personal note, it seems you know more about open-source and package maintenance than I do. I'll contact you via email so we can discuss this more.
